compiler/damlc/lib/DA/Cli/Damlc/Command/Damldoc.hs

```diff
@@ -62,33 +67,44 @@ documentation = Damldoc
             <> help "Name of package to generate."
             <> long "package-name"
 
-    optPrefix :: Parser (Maybe FilePath)
-    optPrefix =
+    optTemplate :: Parser (Maybe FilePath)
+    optTemplate =
         optional . option str
             $ metavar "FILE"
-            <> help "File to prepend to all generated files"
-            <> long "prefix"
-            <> short 'p'
+            <> help "Path to output template for generated files. When generating docs, __TITLE__ and __BODY__ in the template are replaced with doc title and body respectively, before output. (Exception: for hoogle and json output, the template file is a prefix to the body, no replacement occurs.)" -- TODO: make template behavior uniform accross formats
+            <> long "template"
+            <> short 't'
 
     argMainFiles :: Parser [FilePath]
     argMainFiles = some $ argument str $ metavar "FILE..."
                   <> help "Main file(s) (*.daml) whose contents are read"
 
-    optJsonOrFormat :: Parser DocFormat
-    optJsonOrFormat = fromMaybe <$>
-                      optFormat <*>
-                      (flag Nothing (Just Json) $
-                        long "json"
-                        <> help "alias for `--format Json'")
-
-    optFormat :: Parser DocFormat
-    optFormat = option auto $ metavar "FORMAT"
-                <> help ("Output format. Valid format names: "
-                         <> show [minBound..maxBound::DocFormat]
-                         <> " (Default: Markdown).")
-                <> short 'f'
-                <> long "format"
-                <> value Markdown
+    optOutputFormatOrJson :: Parser OutputFormat
+    optOutputFormatOrJson = fromMaybe
+        <$> optOutputFormat
+        <*> (flag Nothing (Just OutputJson) $
+            long "json"
+            <> help "alias for `--format json'")
+
+    optOutputFormat :: Parser OutputFormat
+    optOutputFormat =
+        option readOutputFormat $
+            metavar "FORMAT"
+            <> help "Output format. Valid format names: rst, md, markdown, html, hoogle, json (Default: markdown)."
+            <> short 'f'
+            <> long "format"
+            <> value (OutputDocs Markdown)
+
+    readOutputFormat =
+        eitherReader $ \arg ->
+            case lower arg of
+                "rst" -> Right (OutputDocs Rst)
```

Comment:
Thanks! I kept getting the format names wrong :slightly_smiling_face:

Reply:
:-)